### PR TITLE
Refactor wasm initialization

### PR DESCRIPTION
**Pull request by martinRenou**
This will:
- initialize the WASM just once
- allow passing the wasmPath from outside (allow the webpack notation `import new URL('path-to.wasm', import.meta.url);`)

Note that it makes the library closer to a setup like gdal3 wasm for example: https://github.com/bugra9/gdal3.js/blob/master/src/index.js

TODO still:
- update README
- update tests

### CODE REVIEW

**Comment by martinRenou** [comment]
Don't need this actually. Let's close.

File: src/index.ts [hunk 1]
```diff
@@ -11,141 +11,138 @@ const fetchByteArray = async (url: string): Promise<Uint8Array> => {
...
-        'Extraction failed with status:',
-        statusPtr,
-        'Error:',
-        errorMessage
+  const extractData = async (data: Uint8Array): Promise<IFileData[]> => {
+    /**Since WebAssembly, memory is accessed using pointers
+      and the first parameter of extract_archive method from unpack.c, which is Uint8Array of file data, should be a pointer
+      so we have to allocate memory for file data
+    **/
+    const inputPtr = wasmModule._malloc(data.length);
```
[the remark this answers is not in the document]
Comment (martinRenou) [reply src/index.ts:26]: It's in its scope so it knows about it

File: src/helper.ts [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
-import unpackWasm from './unpack.wasm';
 
-const initializeWasm = async (): Promise<IWasmModule | undefined> => {
+const initializeWasm = async (wasmPath: string | undefined): Promise<IWasmModule> => {
   try {
     const wasmModule: IWasmModule = await unpack({
       locateFile(path: string) {
         if (path.endsWith('.wasm')) {
-          return unpackWasm;
+          return wasmPath ? wasmPath : 'unpack.wasm';
```
[the remark this answers is not in the document]
Comment (martinRenou) [reply src/helper.ts]: This code change has been discarded, I don't need it actually. Thanks!

File: README.md [hunk 3]
```diff
@@ -18,7 +18,9 @@ This package has 2 methods:
 
 The example of using:
 ```sh
-import untarjs from "@emscripten-forge/untarjs";
+import { initUntarJS } from "@emscripten-forge/untarjs";
+
+const untarjs = await initUntarJS();
```
Comment (martinRenou) [inline README.md:23]: @AnastasiaSliusar note the change in the API here, this allows loading the WASM once.

File: src/index.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
 import initializeWasm from './helper';
 import { IFileData } from './types';
-import { IWasmModule } from './unpack';
+
+export interface IUnpackJSAPI {
```
Comment (AnastasiaSliusar) [inline src/index.ts:4]: Hello @martinRenou good work!

I have a small suggestion. I would like to suggest to move IUnpackJSAPI to types.ts or IFileData interface to move from types.ts to index.ts and export both of them. What do you think?

**Comment by martinRenou** [comment]
Thanks for the review @AnastasiaSliusar I've applied your suggestion!

**Pull request closed without merging**